src/ansys/templates/python/solution/{{cookiecutter.__project_name_slug}}/.pre-commit-config.yaml

```diff
@@ -1,28 +1,28 @@
 repos:
 
   - repo: https://github.com/psf/black
-    rev: 22.3.0
+    rev: 24.1.1
```

Comment:
All of these packages, and the exact versions, should be defined in the TOML file as dev dependencies

Reply:
Why do we need this? Pre-commit handles those dependencies right?

Reply:
True, but my concern is mostly for a faster feedback loop. For example, VSCode uses the built venv when you are working on a solution - if things like black/pep8 are installed in the venv, then VSCode will give you immediate feedback when writing code that there are styling issues with your code.

For me, the hooks are almost too late. People generally get frustrated when they write a bunch of code and then find that the hooks (that they usually don't understand) are failing.

I assume similar behavior exists for other code editors if the underlying packages exist in the venv. 

It is out-of-scope for this PR, but I also envision us adding a default/suggested list of VSCode extensions - this can be done via `.vscode/extensions.json` and/or using editorconfig, which is more general.

Reply:
I understand. So I added a dependency group with the same tools listed in pre-commit.-config. I also added a extensions.json with a first set of recommended extensions.